pySDC/core/ConvergenceController.py

```diff
@@ -0,0 +1,70 @@
+from pySDC.helpers.pysdc_helper import FrozenClass
+
+
+# short helper class to add params as attributes
+class _Pars(FrozenClass):
+    def __init__(self, params):
+        self.order = 0
+
+        for k, v in params.items():
+            setattr(self, k, v)
+
+        self._freeze()
+
+
+class ConvergenceController(object):
+    """
+    Base abstract class for convergence controller, which is plugged into the controller to determine the iteration
+    count and time step size.
+    """
+
+    def __init__(self, controller, description):
+        self.params = _Pars(description.get('convergence_control_params', {}))
+        params_ok, msg = self.check_parameters(controller, description)
+        assert params_ok, msg
+        self.reset_global_variables(controller)
+
```

Comment:
Do people HAVE to implement any of the following functions? If so, use `raise NotImplementedError` instead of `pass`.